cmd/snap-seccomp/main.go

```diff
@@ -0,0 +1,522 @@
+
+	"github.com/snapcore/snapd/arch"
+)
+
+// libseccomp maximum per ARG_COUNT_MAX in src/arch.h
+const ScArgsMaxlength = 6
+
+var seccompResolver = map[string]uint64{
+	// man 2 socket - domain and man 5 apparmor.d. AF_ and PF_ are
+	// synonymous in the kernel and can be used interchangeably in
+	// policy (ie, if use AF_UNIX, don't need a corresponding PF_UNIX
+	// rule). See include/linux/socket.h
+	"AF_UNIX":       syscall.AF_UNIX,
+	"PF_UNIX":       C.PF_UNIX,
+	"AF_LOCAL":      syscall.AF_LOCAL,
+	"PF_LOCAL":      C.PF_LOCAL,
+	"AF_INET":       syscall.AF_INET,
+	"PF_INET":       C.PF_INET,
+	"AF_INET6":      syscall.AF_INET6,
+	"PF_INET6":      C.PF_INET6,
+	"AF_IPX":        syscall.AF_IPX,
+	"PF_IPX":        C.PF_IPX,
+	"AF_NETLINK":    syscall.AF_NETLINK,
+	"PF_NETLINK":    C.PF_NETLINK,
+	"AF_X25":        syscall.AF_X25,
+	"PF_X25":        C.PF_X25,
+	"AF_AX25":       syscall.AF_AX25,
+	"PF_AX25":       C.PF_AX25,
+	"AF_ATMPVC":     syscall.AF_ATMPVC,
+	"PF_ATMPVC":     C.PF_ATMPVC,
+	"AF_APPLETALK":  syscall.AF_APPLETALK,
+	"PF_APPLETALK":  C.PF_APPLETALK,
+	"AF_PACKET":     syscall.AF_PACKET,
+	"PF_PACKET":     C.PF_PACKET,
+	"AF_ALG":        syscall.AF_ALG,
+	"PF_ALG":        C.PF_ALG,
+	"AF_BRIDGE":     syscall.AF_BRIDGE,
+	"PF_BRIDGE":     C.PF_BRIDGE,
+	"AF_NETROM":     syscall.AF_NETROM,
+	"PF_NETROM":     C.PF_NETROM,
+	"AF_ROSE":       syscall.AF_ROSE,
+	"PF_ROSE":       C.PF_ROSE,
+	"AF_NETBEUI":    syscall.AF_NETBEUI,
+	"PF_NETBEUI":    C.PF_NETBEUI,
+	"AF_SECURITY":   syscall.AF_SECURITY,
+	"PF_SECURITY":   C.PF_SECURITY,
+	"AF_KEY":        syscall.AF_KEY,
+	"PF_KEY":        C.PF_KEY,
+	"AF_ASH":        syscall.AF_ASH,
+	"PF_ASH":        C.PF_ASH,
+	"AF_ECONET":     syscall.AF_ECONET,
+	"PF_ECONET":     C.PF_ECONET,
+	"AF_SNA":        syscall.AF_SNA,
+	"PF_SNA":        C.PF_SNA,
+	"AF_IRDA":       syscall.AF_IRDA,
+	"PF_IRDA":       C.PF_IRDA,
+	"AF_PPPOX":      syscall.AF_PPPOX,
+	"PF_PPPOX":      C.PF_PPPOX,
+	"AF_WANPIPE":    syscall.AF_WANPIPE,
+	"PF_WANPIPE":    C.PF_WANPIPE,
+	"AF_BLUETOOTH":  syscall.AF_BLUETOOTH,
+	"PF_BLUETOOTH":  C.PF_BLUETOOTH,
+	"AF_RDS":        syscall.AF_RDS,
+	"PF_RDS":        C.PF_RDS,
+	"AF_LLC":        syscall.AF_LLC,
+	"PF_LLC":        C.PF_LLC,
+	"AF_TIPC":       syscall.AF_TIPC,
+	"PF_TIPC":       C.PF_TIPC,
+	"AF_IUCV":       syscall.AF_IUCV,
+	"PF_IUCV":       C.PF_IUCV,
+	"AF_RXRPC":      syscall.AF_RXRPC,
+	"PF_RXRPC":      C.PF_RXRPC,
+	"AF_ISDN":       syscall.AF_ISDN,
+	"PF_ISDN":       C.PF_ISDN,
+	"AF_PHONET":     syscall.AF_PHONET,
+	"PF_PHONET":     C.PF_PHONET,
+	"AF_IEEE802154": syscall.AF_IEEE802154,
+	"PF_IEEE802154": C.PF_IEEE802154,
+	"AF_CAIF":       syscall.AF_CAIF,
+	"PF_CAIF":       C.AF_CAIF,
+	"AF_NFC":        C.AF_NFC,
+	"PF_NFC":        C.PF_NFC,
+	"AF_VSOCK":      C.AF_VSOCK,
+	"PF_VSOCK":      C.PF_VSOCK,
+	// may not be defined in socket.h yet
+	"AF_IB":   C.AF_IB, // 27
+	"PF_IB":   C.PF_IB,
+	"AF_MPLS": C.AF_MPLS, // 28
+	"PF_MPLS": C.PF_MPLS,
```

Comment:
Oh, I missed that this and the ones I commented on later are handled by the preamble before `import "C"`. I'll leave this comment for posterity, but delete the others for clarity.